proposals/12147-mlmd-removal/protos/artifacts.proto

```diff
@@ -0,0 +1,258 @@
+  // Required. Task UUID that owns this metric
+  string task_id = 1;
+  // Required. Name of the metric
+  string name = 2;
+}
+
+message ListMetricsRequest {
+  // A page token to request the results page.
+  string page_token = 1;
+  // The number of metrics to be listed per page.
+  int32 page_size = 2;
+  // Sorting order in form of "field_name", "field_name asc" or "field_name desc".
+  string sort_by = 3;
+  // A url-encoded, JSON-serialized filter protocol buffer.
+  string filter = 4;
+}
+
+message ListMetricsResponse {
+  // The list of metrics returned.
+  repeated Metric metrics = 1;
+  // The total number of metrics available.
+  int32 total_size = 2;
+  // Token to retrieve the next page of results.
+  string next_page_token = 3;
+}
+
+enum ArtifactEventType {
+  INPUT = 0;
+  OUTPUT = 1;
+}
+message ArtifactEvents {
+  string id = 1;
+  string artifact_id = 2;
+  string task_id = 3;
+
+  ArtifactEventType type = 4;
+  google.protobuf.Timestamp created_at = 5;
+}
+
+enum MetricType {
+  METRIC_INPUT = 0;
+  METRIC_OUTPUT = 1;
+}
+
+message Metric {
+  // Required. Task UUID that owns this metric
+  string task_id = 1;
+  // Required. Name of the metric
+  string name = 2;
+  // Optional. Numeric value of the metric
+  double number_value = 3;
+  // Optional. JSON value of the metric
+  string json_value = 4;
+  // Required. Schema of the metric
+  enum Schema {
+      Metric = 0;
+      ClassificationMetric = 1;
+      SlicedClassificationMetric = 2;
+  }
+  Schema schema = 5;
+  google.protobuf.Timestamp created_at = 6;
+  // Required. Type of the metric (input/output)
+  MetricType type = 7;
+}
+
+// Note to be confused with RuntimeArtifact in pipelinespec
+message Artifact {
+  // Output only. The unique server generated id of the artifact.
+  // Note: Updated id name to be consistent with other api naming patterns (with prefix)
+  string artifact_id = 1;
+  // The client provided name of the artifact.
+  // Note: it seems in MLMD when name was set, it had to be unique for that type_id
+  // this restriction is removed here
+  string name = 2;
+
+  enum ArtifactType {
+    Artifact = 0;
+    Model = 1;
+    Dataset = 2;
+    HTML = 3;
+    Markdown = 4;
+  }
+  // The name of an ArtifactType. E.g. Dataset
+  ArtifactType type = 3;
+
+  // The uniform resource identifier of the physical artifact.
+  // May be empty if there is no physical artifact.
+  string uri = 4;
+  // User provided custom properties which are not defined by its type.
+  map<string, Value> custom_properties = 5;
+  // Output only. Create time of the artifact in millisecond since epoch.
+  // Note: The type and name is updated from mlmd artifact to be consistent with other backend apis.
+  google.protobuf.Timestamp created_at = 6;
+
+  // New field:
+  string namespace = 7;
+
+  // In KFP only the Live state is ever used
+  // Optionally we can omit this and always assume
+  // an artifact fetched is "Live" and add a state
+  // later if needed
+  enum State {
```

Comment:
I don't like how the state is different than `RuntimeState` in `run.proto`. Could we consider aligning these with an optional field of `mlmdState` to map to the old verbiage?